agents/agent.py

```diff
@@ -721,3 +721,130 @@ def maybe_add_to_queue(self, data):
 
         if command_history.command in self.whitelist_commands:
             self.queue.put(command_history)
+
+    def get_cpu_usage(self, interval=60):
+        """
+        Get the average CPU usage percentage over the last minute.
+        """
+        try:
+            return psutil.cpu_percent(interval=interval)
+        except Exception as e:
```

Comment:
n the majority of cases including this one catching generic Exception is not a good practice.